packages/edit-post/src/components/sidebar/post-link/index.js

```diff
@@ -0,0 +1,142 @@
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import { PanelBody, TextControl } from '@wordpress/components';
+import { withSelect, withDispatch } from '@wordpress/data';
+import { compose, ifCondition, withState } from '@wordpress/compose';
+import { addQueryArgs } from '@wordpress/url';
+
+/**
+ * Module Constants
+ */
+const PANEL_NAME = 'post-link';
+
+function PostLink( {
+	isOpened,
+	onTogglePanel,
+	isEditable,
+	postLink,
+	permalinkParts,
+	editPermalink,
+	postType,
+	forceEmptyField,
+	setState,
+} ) {
+	const { prefix, postName, suffix } = permalinkParts;
+	const prefixElement = prefix && (
+		<span className="edit-post-post-link__link-prefix">{ prefix }</span>
+	);
+	const postNameElement = postName && (
+		<span className="edit-post-post-link__link-post-name">{ postName }</span>
+	);
+	const suffixElement = suffix && (
+		<span className="edit-post-post-link__link-suffix">{ suffix }</span>
+	);
+
+	const singularLabel = get( postType, [ 'labels', 'singular_name' ] );
+	return (
+		<PanelBody
+			title={
+				// translators: %s: post type sigular name label e.g: Post, Page etc...
+				sprintf( __( '%s Link' ), singularLabel )
+			}
+			opened={ isOpened }
+			onToggle={ onTogglePanel }
+		>
+			<TextControl
+				label={
+					// translators: %s: post type sigular name label e.g: Post, Page etc...
+					sprintf( __( '%s URL' ), singularLabel )
+				}
+				value={ forceEmptyField ? '' : postName }
+				readonly={ ! isEditable }
+				onChange={ ( newValue ) => {
+					editPermalink( newValue );
+					// When we delete the field the permalink gets
+					// reverted to the original value.
+					// The forceEmptyField logic allows the user to have
+					// the field temporarily empty while typing.
+					if ( ! newValue ) {
+						if ( ! forceEmptyField ) {
+							setState( {
+								forceEmptyField: true,
+							} );
+						}
+						return;
+					}
+					if ( forceEmptyField ) {
+						setState( {
+							forceEmptyField: false,
+						} );
+					}
+				} }
+			/>
+			<p className="edit-post-post-link__preview-label">
+				{ __( 'Preview' ) }
+			</p>
+			<a className="edit-post-post-link__link" href={ postLink }>
```

Comment:
```suggestion
			<a className="edit-post-post-link__link" href={ postLink } target="_blank">
```
The preview should open in a new window.